### PR TITLE
Add factory method for log events to assert on for better readability

### PR DESCRIPTION
**Pull request by logi**
I have made a number of additions to pytest-structlog locally and this is the lowest-hanging fruit of that to submit. It makes the lists of log events to assert on more terse and more readable.

### CODE REVIEW

**Comment by logi** [comment]
FYI, I have 2 other branches that I'll submit PRs from, each building on the last. You can see them here:
- https://github.com/logi/pytest-structlog/tree/filter-event-lists
- https://github.com/logi/pytest-structlog/tree/match-partial-events-in-lists

**Comment by logi** [comment]
PS If you're at PyCon I'll be happy to talk about this at Squatters.

File: pytest_structlog.py [hunk 1]
```diff
@@ -61,6 +73,30 @@ def has(self, message, **context):
         context["event"] = message
         return any(is_submap(context, e) for e in self.events)
 
+    def log(self, level, event, **kw):
+        """Create log event to assert against"""
+        return {"level": level_to_name(level), "event": event, **kw}
```
Comment (wimglenn) [inline pytest_structlog.py:78]: For older Python versions compat 

```suggestion
        result = {"level": level_to_name(level), "event": event}
        result.update(kw)
        return result
```

File: pytest_structlog.py [hunk 2]
```diff
@@ -37,6 +39,16 @@ def __lt__(self, other):
 absent = object()
 
 
+def level_to_name(level):
+    """Given the name or number for a log-level, return the lower-case level name."""
+    if isinstance(level, str):
+        return level.lower()
+    try:
+        return logging._levelToName[level].lower()
```
Comment (wimglenn) [inline pytest_structlog.py:47]: It looks like [`logging.getLevelName`](https://docs.python.org/3/library/logging.html#logging.getLevelName) is the public API here.

Comment (logi) [reply pytest_structlog.py:47]: Ah, I wasn't aware of that!

I also seem to have broken the tests in 2.7 for no good reason.

Comment (logi) [reply pytest_structlog.py:47]: That function seems to be a minor shitshow but it should be safe to pass integers into it.

> Changed in version 3.4: In Python versions earlier than 3.4, this function could also be passed a text level, and would return the corresponding numeric value of the level. This undocumented behaviour was considered a mistake, and was removed in Python 3.4, but reinstated in 3.4.2 due to retain backward compatibility.

**Comment by logi** [comment]
Any further thoughts about this PR?

Since I've got it open, here is code lifted directly from one of our unit tests. I especially like the last line but that's for another PR...
```python
assert log.events >= [
    log.info(
        "Processing started for calibration",
        calibration_id=2,
    ),
    log.info(
        "Request measurement data",
        auth_user="test_mms_user",
        end_time=fix_time("2018-09-03T18:30"),
        response_code=200,
        sensor_ids=["07000997:DE:0"],
        start_time=fix_time("2018-09-02T18:00"),
    ),
    log.info(
        "Request measurement data",
        auth_user="test_mms_user",
        end_time=fix_time("2018-09-03T18:30"),
        response_code=200,
        sensor_ids=["10000994:DE:0"],
        start_time=fix_time("2018-09-02T18:00"),
    ),
    log.info(
        "Found stable range for setpoint in period",
        end_time=fix_time("2018-09-02T18:40"),
        reference_sensor_id="07000997:DE:0",
        setpoint=25.0,
        setpoint_average=Decimal("25.1"),
        setpoint_id=5,
        start_time=fix_time("2018-09-02T18:00"),
    ),
    log.debug(
        "Setpoint passed",
        average=Decimal("25.6"),
        average_temperature_error=Decimal("0.5"),
        end_time=fix_time("2018-09-02T18:40"),
        passed=True,
        reference_sensor_id="07000997:DE:0",
        start_time=fix_time("2018-09-02T18:00"),
        sensor_id="10000994:DE:0",
        setpoint=25.0,
        setpoint_average=Decimal("25.1"),
        setpoint_id=5,
        tolerance=Decimal("0.5000"),
    ),
    log.info(
        "Processing of calibration completed successfully",
        calibration_id=2,
    ),
]
assert not log.events.warnings()

```

**Comment by wimglenn** [comment]
@logi  This looks OK to me! My only concern is that pytest-structlog was, previously, blissfully unaware of stdlib logging. It was "pure structlog" so to speak. However, there's not really any huge gain from that, and anyway structlog already imports stdlib logging these days regardless.

Note that the trick you've used [here](https://github.com/logi/pytest-structlog/blob/a42fa1f2f355ad092ff05fef39a8ff40c4e16ecd/pytest_structlog.py#L77) has been called [despicable](https://mail.python.org/pipermail/python-dev/2010-April/099459.html) by Guido van Rossum.

**Pull request merged**